File: baekjoon/10828.py
import sys

stack = []
count  = 0
command_num = int(sys.stdin.readline())
while True:
    if count == command_num:
        break
    command = sys.stdin.readline().split()
    if command[0] =="push":
        stack.append(int(command[1]))
    elif command[0] =="pop":
        if len(stack) == 0:
            print(-1)
        else:
            print(stack[-1])
            if len(stack) == 1:
                stack = []
            else:
                stack = stack[:-1]
    elif command[0] =="size":
        print(len(stack))
    elif command[0] =="empty":
        if len(stack) == 0:
            print(1)
        else:
            print(0)
    elif command[0] =="top":
        if len(stack) == 0:
            print(-1)
        else:
            print(stack[-1])
    count = count + 1

    
        
        

File: baekjoon/10828.cpp
#include <iostream>
#include <stack>

using namespace std;

int main()
{
    int n; // 명령의 갯수
    stack <int> my_stack; // 정수를 저장하는 스택

    cin >> n;
    for (int i = 0; i < n; i++)
    {
        string command;
        int number;

        cin >> command;
        if (command == "push")
        {
            cin >> number;
            my_stack.push(number);
        }
        if (command == "pop")
        {
            if (my_stack.size() == 0)
            {
                cout << -1 << "\n";
                continue;
            }
            cout << my_stack.top() << "\n"; 
            my_stack.pop();
        }
        if (command == "size")
        {
            cout << my_stack.size() << "\n";
        }
        if (command == "empty")
        {
            cout << my_stack.empty() << "\n";
        }
        if (command == "top")
        {
            if (my_stack.size() == 0)
            {
                cout << -1 << "\n";
                continue;
            }
            cout << my_stack.top() << "\n";
        }
    } 
}